src/sandstorm/sandstorm-http-bridge.c++

```diff
@@ -1230,15 +1239,79 @@ public:
     }
   }
 
-  std::map<kj::StringPtr, SessionContext::Client&> sessions;
-  // TODO(cleanup): Make this private with appropriate accessor methods.
+  kj::Maybe<SessionContext::Client&> findSession(const kj::StringPtr& id) {
+    return findInMap(sessions, id);
+  }
+
+  kj::Maybe<SandstormHttpBridge::GetSessionOfferResults::Reader&> findOffer(const kj::StringPtr& id) {
+    KJ_IF_MAYBE(ret, findInMap(offers, id)) {
+      return ret->get();
+    } else {
+      return nullptr;
+    }
+  }
+
+  kj::Maybe<SandstormHttpBridge::GetSessionRequestResults::Reader&> findRequest(const kj::StringPtr& id) {
+    KJ_IF_MAYBE(ret, findInMap(requests, id)) {
+      return ret->get();
+    } else {
+      return nullptr;
+    }
+  }
+
+  void eraseSession(const kj::StringPtr& id) {
+    sessions.erase(id);
+    offers.erase(id);
+    requests.erase(id);
+  }
+
+  void insertSession(const kj::StringPtr& id, SessionContext::Client& session) {
+    sessions.insert({kj::StringPtr(id), session});
+  }
+
+  void insertRequest(
+      const kj::StringPtr& id,
+      capnp::List<PowerboxDescriptor>::Reader descriptor) {
+
+    auto msg = capnp::MallocMessageBuilder();
+    auto results = msg.initRoot<SandstormHttpBridge::GetSessionRequestResults>();
+    results.setRequestInfo(descriptor);
+
+    requests.insert({kj::StringPtr(id), capnp::clone(results.asReader())});
+  }
+
+  void insertOffer(
+      const kj::StringPtr& id,
+      capnp::Capability::Client&& offer,
+      PowerboxDescriptor::Reader descriptor) {
+
+    auto msg = capnp::MallocMessageBuilder();
+    auto results = msg.initRoot<SandstormHttpBridge::GetSessionOfferResults>();
+
+    results.setOffer(offer);
+    results.setDescriptor(descriptor);
+
+    offers.insert({kj::StringPtr(id), capnp::clone(results.asReader())});
+  }
 
 private:
   SandstormApi<BridgeObjectId>::Client apiCap;
   spk::BridgeConfig::Reader config;
   kj::AutoCloseFd identitiesDir;
   kj::AutoCloseFd trashDir;
 
+  // These maps store information relating to active sessions. `sessions` always has an entry
+  // for any active session that is handled by the app. `requests` and `offers` will only
+  // have an entry for request and offer sessions, respectively.
+  //
+  // In all cases the keys are owned by the session object, which is responsible for deleting
+  // the session from these maps (via eraseSession) on cleanup. For `sessions`, the value
+  // is also owned by the session. For `requests` and `offers`, the value is owned by the map
+  // itself.
+  std::map<kj::StringPtr, SessionContext::Client&> sessions;
```

Comment:
Hmm, why parallel maps? Why not have one map to a struct type that includes a `kj::OneOf` of the possible powerbox info?

Reply:
I guess that makes more sense. I'll tweak it. Maybe since I've already got the `SessionInfo` type defined I'll use that instead of a `kj::OneOf`.

Reply:
Done.